src/taint_analysis.cpp

```diff
@@ -49,6 +75,21 @@ void TaintAnalyzer::visit_assignment(AssignmentNode<std::set<std::string>> &node
     }
 }
 
+void TaintAnalyzer::visit_arrayAssignment(ArrayAssignmentNode<std::set<std::string>> &node)
+{
+    node.state = join(node);
+
+    if (evaluateExpression(node.expression, node.state))
+    {
+        node.state.insert(node.id);
+    }
+    else
+    {
+        node.state.erase(node.id);
+    }
```

Comment:
I think this case is missing some aspects. If x[2] = {£, 5}; is altered x[1]=5; because 5 is not tainted this if statement would evaluate to false and the whole array would be set to untainted despite x[0]=£

Reply:
removed individual index tracking, now the array will stay tainted if it gets tainted somehow